cmd/snap-seccomp/old_seccomp.go

```diff
@@ -0,0 +1,31 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+// +build oldseccomp
```

Comment:
hmm this is an interesting way to solve the problem, I think our original idea was to have the packaging scripts for trusty instead add a new line to go.mod like this:

```
replace github.com/seccomp/libseccomp-golang => github.com/mvo5/libseccomp-golang
```

I'm curious if you looked at this at all? What you have is probably fine, but is slightly more opaque than a packaging script modifying go.mod here I think.

Reply:
I couldn't make `go.mod` to be universal for both upstream and trusty at the same time. I.e. `go mod tidy` would continiously drop either mvo5 or the seccomp version of the library. Leaving a gofile with an unused build tag was the only way i figured out to keep `go mod tidy` with both editions of the library.

Reply:
I didn't mean to have a universal go.mod, I just meant to modify the go.mod for trusty alone when building the deb for trusty with like `echo "replace ... " >> go.mod` in the `dh_override_whatever`

Reply:
oh, interesting. I can try hacking that.

Reply:
my concern was to ensure that both copies of vendored lib are available whenever one does snapd-vendor style packaging.

Reply:
hmm, not sure that is super relevant since we mainly just do that for other distros where they don't need the fork of seccomp as they have it packaged already

Reply:
it's exactly how we upload ubuntu-14.04 though, no?

Reply:
uhh I'm confused now, what do you mean about uploading ubuntu-14.04 with vendor sources?

Reply:
```
pull-lp-source snapd trusty
ls snapd-2.38~14.04/vendor/github.com/mvo5/libseccomp-golang/
```

Something, somewhere, when building 3.0 (native) trusty package downloads all sources when creating source package. Then it is uploaded to launchpad to build. During the build in launchpad it has no network access to github.com to download anything. Meaning to avoid breaking/changing how source package preparation is done, i should ensure that go.mod references mvo5 fork, and it ends up in the source tarballs.

I am also discovering that whilst patches are applied for debian-sid source package builds, they are not applied for the native builds. Thus I will also be trying out to use the `replace` directive during build. Cause i think it is easier than patching; and it is a simple one liner.

Reply:
> echo "replace ... " >> go.mod

FWIW this should probably be:
```
go mod edit -replace=github.com/seccomp/libseccomp-golang=github.com/mvo5/libseccomp-golang
```